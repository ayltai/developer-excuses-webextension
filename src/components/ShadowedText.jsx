import { makeStyles, Typography, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

const RawShadowedText = props => {
    const classes = makeStyles(() => ({
        text : {
            fontFamily : props.fontFamily,
            fontWeight : props.fontWeight || 400,
            fontSize   : props.fontSize || 12,
            textShadow : `0 0 ${props.fontShadow * 2 + (props.fontShadow > 0 ? 1 : 0)}px rgba(0, 0, 0, 0.85)`,
            cursor     : props.onClick ? 'pointer' : 'default',
        },
    }))();

    return (
        <Typography
            className={props.className}
            color='textPrimary'>
            <div
                className={classes.text}
                onClick={() => {
                    if (props.onClick) props.onClick();
                }}>
                {props.text}
            </div>
        </Typography>
    );
};

export const ShadowedText = React.memo(RawShadowedText);

ShadowedText.propTypes = {
    className  : PropTypes.string,
    text       : PropTypes.string,
    fontFamily : PropTypes.string,
    fontWeight : PropTypes.number,
    fontSize   : PropTypes.number,
    fontShadow : PropTypes.number,
    onClick    : PropTypes.func,
};

RawShadowedText.propTypes = ShadowedText.propTypes;
