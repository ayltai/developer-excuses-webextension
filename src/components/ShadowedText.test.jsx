import { mount, shallow, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { ShadowedText, } from './ShadowedText';

describe('<ShadowedText />', () => {
    it('renders correctly', () => {
        expect(renderer.create(
            <ShadowedText
                text='dummy'
                fontFamily='Arial'
                fontWeight={400}
                fontSize={16}
                fontShadow={2}
                onClick={() => {}} />
        ).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <ShadowedText
                text='dummy'
                fontWeight={400} />
        );
    });

    it('triggers onClick when clicked', () => {
        const onClick = jest.fn();

        shallow(
            <ShadowedText
                text='dummy'
                fontFamily='Arial'
                fontSize={16}
                fontShadow={2}
                onClick={onClick} />)
            .find('div')
            .simulate('click');

        expect(onClick.mock.calls.length).toEqual(1);
    });

    it('does not trigger onClick when clicked', () => {
        shallow(
            <ShadowedText
                text='dummy'
                fontFamily='Arial'
                fontSize={16}
                fontShadow={2} />)
            .find('div')
            .simulate('click');
    });
});
