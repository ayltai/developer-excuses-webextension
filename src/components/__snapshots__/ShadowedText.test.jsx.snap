// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShadowedText /> renders correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
>
  <div
    className="makeStyles-text-1"
    onClick={[Function]}
  >
    dummy
  </div>
</p>
`;
